webapp/components/admin_console/login_settings.jsx

```diff
@@ -74,6 +139,103 @@ export default class LoginSettings extends AdminSettings {
 
         return (
             <SettingsGroup>
+                <TextSetting
+                    id='passwordMinimumLength'
+                    label={
+                        <FormattedMessage
+                            id='admin.password.minimumLength'
+                            defaultMessage='Minimum Password Length:'
+                        />
+                    }
+                    helpText={
+                        <FormattedMessage
+                            id='admin.password.minimumLengthDescription'
+                            defaultMessage='Minimum number of characters required for a valid password.'
+                        />
+                    }
+                    value={this.state.passwordMinimumLength}
+                    onChange={this.handleChange}
+                />
+                <Setting
```

Comment:
I'd move this all into it's own component and pass in the state containing the config. I know it won't be reused anywhere else, but it'll make it tidier and more self-contained. That and you'll actually be able to use `this.handleChange` to properly set the state so that you don't need refs. If you look at the other Setting components, their onChange is called differently than the default one used by regular HTML inputs.